vmdb/app/controllers/application_controller.rb

```diff
@@ -762,10 +762,14 @@ def disable_client_cache
   end
 
   # Common method enable/disable schedules
-  def schedule_enable_disable(schedules, task)
-    MiqSchedule.find_all_by_id(schedules, :order => "lower(name)").each do |schedule|
-      schedule.enabled = (task == "enable")
-      schedule.save!
+  def schedule_enable_disable(schedules, enabled)
+    MiqSchedule.where(:id => schedules).order("lower(name)").each do |schedule|
+      if schedule.enabled == enabled
+        schedule.touch
```

Comment:
I'm fine with the rest of the changes.  I think we can simplify this significantly by only updating the enabled column if it's changed.